app/main.py

```diff
@@ -1,6 +1,19 @@
 from typing import Callable
+import functools
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+
+    decorator_cache = {}
+
+    @functools.wraps(func)
+    def wrapper(*args) -> int:
+        if args in decorator_cache:
+            print("Getting from cache")
+            return decorator_cache[args]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            decorator_cache[args] = result
+            return result
```

Comment:
Try to rewrite your decorator not to have return duplicated. You can put return after a conditional statement